actions/websocket_actions.jsx

```diff
@@ -524,10 +525,36 @@ function handlePreferencesDeletedEvent(msg) {
 }
 
 function handleUserTypingEvent(msg) {
-    GlobalActions.emitRemoteUserTypingEvent(msg.broadcast.channel_id, msg.data.user_id, msg.data.parent_id);
+    const state = getState();
```

Comment:
why are you duplicating code with mattermost-redux on this? https://github.com/mattermost/mattermost-redux/blob/master/src/actions/websocket.js#L620

Reply:
ok so I see the websocket client can only take a one event handler callback...         websocketClient.setEventCallback(handleEvent);
we should change it to allow more than 1 callback and only leave webapp-specific code inside websocket_actions(that redux is not taking care of); really don't like the way it is currently duplicated

Reply:
Us neither, the websocket client code is duplicated in the mattermost-redux and mattermost-webapp, we have to find the time to migrate it to use only the mattermost-redux and remove entirely the websocket client from the mattermost-webapp. Until them, I had to replicate the behavior in the mattermost-webapp repo.

Reply:
The ticket to work on that is here if anyone wants to give it a shot :) It's something I've wanted to get done for a while, just haven't had the time
https://mattermost.atlassian.net/browse/MM-8648